metricflow/test/generate_snapshots.py

```diff
@@ -0,0 +1,112 @@
+"""Script to help generate test snapshots for all supported engines.
+
+Credentials are stored as a JSON string in an environment variable set via a shell command like:
+
+export MF_TEST_ENGINE_CREDENTIALS=$(cat <<EOF
+{
+    "duck_db": {
+        "engine_url": null,
+        "engine_password": null
+    },
+    "postgres": {
+        "engine_url": "postgresql://...",
+        "engine_password": "..."
+    },
+    "redshift": {
+        "engine_url": "redshift://...",
+        "engine_password": "..."
+    },
+    "snowflake": {
+        "engine_url": "snowflake://...",
+        "engine_password": "..."
+    },
+    "big_query": {
+        "engine_url": "bigquery://",
+        "engine_password": "..."
+    },
+    "databricks": {
+        "engine_url": "databricks://...",
+        "engine_password": "..."
+    }
+}
+EOF
+)
+"""
+from __future__ import annotations
+
+import logging
+import os
+from typing import Optional, Sequence
+
+from dbt_semantic_interfaces.implementations.base import FrozenBaseModel
+
+from metricflow.configuration.env_var import EnvironmentVariable
+
+logger = logging.getLogger(__name__)
+
+
+class MetricFlowTestCredentialSet(FrozenBaseModel):  # noqa: D
+    engine_url: Optional[str]
+    engine_password: Optional[str]
+
+
+class MetricFlowTestCredentialSetForAllEngines(FrozenBaseModel):  # noqa: D
+    duck_db: MetricFlowTestCredentialSet
+    postgres: MetricFlowTestCredentialSet
+    redshift: MetricFlowTestCredentialSet
+    snowflake: MetricFlowTestCredentialSet
+    big_query: MetricFlowTestCredentialSet
+    databricks: MetricFlowTestCredentialSet
+
+    @property
+    def as_sequence(self) -> Sequence[MetricFlowTestCredentialSet]:  # noqa: D
+        return (self.duck_db, self.postgres, self.redshift, self.snowflake, self.big_query, self.databricks)
+
+
+SNAPSHOT_GENERATING_TEST_FILES = (
+    "metricflow/test/dataflow/builder/test_dataflow_plan_builder.py",
+    "metricflow/test/dataflow/optimizer/source_scan/test_cm_branch_combiner.py",
+    "metricflow/test/dataflow/optimizer/source_scan/test_source_scan_optimizer.py",
+    "metricflow/test/integration/test_rendered_query.py",
+    "metricflow/test/plan_conversion/test_dataflow_to_sql_plan.py",
```

Comment:
I think this is the complete set, but we won't know until we try it.

```
$ git grep -l "assert.*snapshot" | sort
metricflow/test/dataflow/builder/test_dataflow_plan_builder.py
metricflow/test/dataflow/optimizer/source_scan/test_cm_branch_combiner.py
metricflow/test/dataflow/optimizer/source_scan/test_source_scan_optimizer.py
metricflow/test/integration/test_rendered_query.py
metricflow/test/model/test_data_warehouse_tasks.py
metricflow/test/plan_conversion/test_dataflow_to_sql_plan.py
metricflow/test/plan_utils.py
metricflow/test/sql/compare_sql_plan.py
metricflow/test/table_snapshot/test_table_snapshots.py
```

Reply:
Updated.